x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -52,12 +66,18 @@ export class InnerJoin {
     return `${this.getSourceDataRequestId()}_${FORMATTERS_DATA_REQUEST_ID_SUFFIX}`;
   }
 
-  getLeftField() {
+  getLeftField(): IField {
+    if (!this._leftField) {
+      throw new Error('Cannot get leftField from InnerJoin with incomplete config');
+    }
     return this._leftField;
   }
 
-  joinPropertiesToFeature(feature, propertiesMap) {
-    const rightMetricFields = this._rightSource.getMetricFields();
+  joinPropertiesToFeature(feature: Feature, propertiesMap: PropertiesMap): boolean {
+    if (!feature.properties || !this._leftField || !this._rightSource) {
+      return false;
+    }
+    const rightMetricFields = this._rightSource!.getMetricFields();
```

Comment:
Not null operator not needed on this line since there is a if check for rightSource above.